container/sagemaker/python_service.py

```diff
@@ -46,23 +46,33 @@ def __init__(self):
         self._tfs_grpc_port = os.environ['TFS_GRPC_PORT']
         self._tfs_rest_port = os.environ['TFS_REST_PORT']
 
-        self._handler, self._input_handler, self._output_handler = self._import_handlers()
-        self._handlers = self._make_handler(self._handler,
-                                            self._input_handler,
-                                            self._output_handler)
-
-    def on_post(self, req, res):
-        data, context = self._parse_request(req)
+    def on_post(self, req, res, model_name=None):
         try:
-            res.status = falcon.HTTP_200
-            res.body, res.content_type = self._handlers(data, context)
+            if model_name:
+                res.status = falcon.HTTP_200
+                res.body = self._dynamic_endpoint_invocation(req, model_name)
+            else:
+                self._handler, self._input_handler, self._output_handler = self._import_handlers()
+                self._handlers = self._make_handler(self._handler,
+                                                    self._input_handler,
+                                                    self._output_handler)
+                data, context = self._parse_request(req)
+                res.status = falcon.HTTP_200
+                res.body, res.content_type = self._handlers(data, context)
         except Exception as e:  # pylint: disable=broad-except
             log.exception('exception handling request: {}'.format(e))
             res.status = falcon.HTTP_500
             res.body = json.dumps({
                 'error': str(e)
             }).encode(encodings.utf_8.getregentry().name)  # pylint: disable=E1101
 
+    def _dynamic_endpoint_invocation(self, req, model_name):
```

Comment:
if there are no custom python handlers, then why send these requests through the python service at all? performance will be *much* better if we use nginx config to route the requests directly to tfs.

Reply:
If we directly route '/models/.*/invoke' to 'js_content invocations', we cannot use dynamic endpoint and pre/post-processing at the same time. 

And if we use `%FORWARD_INVOCATION_REQUESTS%;` (as we had in '/invocations') to allow both js_content and pre/post-processing, we got this error, since forwarding requests to python service uses `proxy_pass` ["proxy_pass" cannot have URI part in location given by regular expression, or inside named location, or inside "if" statement, or inside "limit_except" block]. 

That's why I'm adding route in the python service.